.github/workflows/tiobe-tics-cron.yaml

```
@@ -0,0 +1,101 @@
+name: TiCS Nightly Security Report
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 10 * * *'
+
+permissions:
+  contents: read
+
+jobs:
+  TiCS:
+
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          # Latest branches
+          - { branch: main }
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@v2
+        with:
+          egress-policy: audit
+
+      - name: Checking out repo
+        uses: actions/checkout@v4
+        with:
+          ref: ${{matrix.branch}}
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install Python Testing Deps
+        run: |
+          set -eux -o pipefail
+
+          # Upgrade pip to be able to read the requirements.txt
+          sudo python3 -m pip install --upgrade pip
+
+          # tox required for running the unit tests with coverage:
+          sudo python3 -m pip install tox
+
+          # Required by TICSQServer:
+          sudo python3 -m pip install pylint flake8
+
+          # Must pre-install dependencies for TICSQServer:
+          sudo python3 -m pip install -r $GITHUB_WORKSPACE/charms/worker/k8s/requirements.txt
+          sudo python3 -m pip install -r $GITHUB_WORKSPACE/test_requirements.txt
```

can we use the `setup-python` action??

https://github.com/actions/setup-python

```suggestion
    - uses: actions/setup-python@v5
      with:
        python-version: '3.12'
        cache: 'pip' # caching pip dependencies
    - name: Install Python Testing Deps
        run: |
          # tox required for running the unit tests with coverage:
          pip install tox

          # Required by TICSQServer:
          pip install pylint flake8

          # Must pre-install dependencies for TICSQServer:
          pip install -r $GITHUB_WORKSPACE/charms/worker/k8s/requirements.txt
          pip install -r $GITHUB_WORKSPACE/test_requirements.txt
```